engine/src/main/java/org/terasology/engine/world/chunks/pipeline/ChunkProcessingPipeline.java

```diff
@@ -141,53 +139,44 @@ private boolean processChunkTasks() {
                     providedChunks.add(chunk);
                 }
 
-                // If another thread is running the task, just skip it
-                if (satisfied && chunkProcessingInfo.lock.tryLock()) {
+                if (satisfied) {
                     try {
-                        try (ThreadActivity ignored = ThreadMonitor.startThreadActivity(chunkTask.getName())) {
-                            chunkProcessingInfo.setChunk(chunkTask.apply(providedChunks));
+                        try (ThreadActivity ignored = ThreadMonitor.startThreadActivity(task.getName())) {
+                            info.setChunk(task.apply(providedChunks));
```

Comment:
I wonder if this can cause a race condition if the chunk is updated while another thread is getting it because it's a requirement of a different chunk. This isn't a new problem, though, it was already there, just obscured by all the locking.

Reply:
only one thread has access to the task.

Reply:
No, `getChunkBy()` gets the task out of the map and gets the chunk from it. The task is in the map the whole time.

Reply:
oh, yea I see, I guess you could put a lock on the chunk() object instead of locking the whole task.